spec-tests/src/test/java/org/candlepin/spec/pools/PoolSpecTest.java

```diff
@@ -0,0 +1,625 @@
+            .findFirst();
+
+        assertTrue(dfilter.isPresent());
+        PoolDTO dpool = dfilter.get();
+
+        // Delete the pool
+        adminClient.pools().deletePool(pool.getId());
+
+        // Verify the both pools were deleted
+        pools = adminClient.pools()
+            .listPools(owner.getId(), null, null, true, null);
+
+        assertNotNull(pools);
+        assertEquals(0, pools.size());
+
+        assertNotFound(() -> adminClient.pools().getPool(npool.getId(), null, null));
+        assertNotFound(() -> adminClient.pools().getPool(dpool.getId(), null, null));
+    }
+
+    @Test
+    @DisplayName("should include calculated attributes when fetching pools")
+    public void shouldIncludeCalculatedAttributesWhenFetchingPools() throws ApiException {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = this.createOwner(adminClient);
+        PoolDTO pool = this.createPool(adminClient, owner);
+
+        ApiClient ownerClient = this.createUserClient(adminClient, owner);
+
+        String consumerName = StringUtil.randomSuffix("test_consumer");
+        ConsumerDTO consumer = this.createConsumer(ownerClient, owner, consumerName, null);
+        ApiClient consumerClient = this.createConsumerClient(ownerClient, consumer);
+
+        PoolDTO fetched = consumerClient.pools().getPool(pool.getId(), consumer.getUuid(), null);
+        assertNotNull(fetched);
+
+        Map<String, String> attributes = fetched.getCalculatedAttributes();
+        assertNotNull(attributes);
+
+        assertThat(attributes)
+            .isNotNull()
+            .hasSizeGreaterThanOrEqualTo(2)
+            .containsEntry("suggested_quantity", "1")
+            .containsEntry("compliance_type", "Standard");
+    }
+
+  // it 'should allow fetching content delivery network by pool id' do
+  //   skip("candlepin running in hosted mode") if is_hosted?
+  //   cdn_label = random_string("test-cdn")
+  //   cdn = create_cdn(cdn_label, "Test CDN", "https://cdn.test.com")
+  //   cdn.id.should_not be nil
+  //   @opts = {"cdn_label"=> cdn_label}
+  //   @cp_export = StandardExporter.new
+  //   @cp_export.create_candlepin_export()
+  //   @cp_export_file = @cp_export.export_filename
+  //   @import_owner = @cp.create_owner(random_string("test_owner"))
+  //   import_record = @cp.import(@import_owner['key'], @cp_export_file)
+  //   pools = @cp.list_owner_pools(@import_owner['key'])
+  //   # only master pools have cdns associated with them
+  //   pools = pools.select do |pool|
+  //       pool['type'] == 'NORMAL'
+  //   end
+  //   pools.each do |pool|
+  //       pool.cdn.should be nil
+  //       result_cdn = @cp.get_cdn_from_pool(pool['id'])
+  //       result_cdn.name.should == cdn.name
+  //       result_cdn.url.should == cdn.url
+  //   end
+  //   @cp.delete_owner(@import_owner['key'])
+  //   @cp_export.cleanup
+  // end
+
+    // @Test
+    // public void shouldAllowFetchingCDNByPoolId() throws ApiException {
```

Comment:
Where are we on the CDN usage investigation? Is it not used at all by either Sat/Hosted?

Reply:
Doesn't seem to be. Next step would be checking with PMs to sort out removal details or if there's some feature or something critical that's on their roadmap.